apps/backend/src/users/users.service.ts

```diff
@@ -157,7 +157,10 @@ export class UsersService {
     }
   }
 
-  async testPassword(updateUserDto: UpdateUserDto, user: User): Promise<void> {
+  async testPassword(
+    updateUserDto: {currentPassword?: string},
+    user: User
+  ): Promise<void> {
```

Comment:
@rbclark 

how do you feel about reuse of testPassword function here? Currently testPassword exists on user.service but now it is also being called as part of apiKey.service. Because of its use in apiKey.service update-api-key.dto includes a currentPassword prop. Now we change the definition of testPassword to not take a user related DTO but just object that has a currentPassword prop.

To me this seems like this more generic implementation should be pulled up to authn service

Reply:
@rbclark